department-of-veterans-affairs/abd-vro
# Prevent duplicate contention migration (fix #2054)

## What was the problem?

The merge app did not check if an EP400 contention already existed in the pending claim before migrating it.

Associated tickets or Slack threads:
- #2054 

## How does this fix it?[^1]

The merge job skips migrating a contention if there is already one in the pending claim with the same `claimantText` and `contentionTypeCode`.

## How to test this PR
- From the `abd-vro` directory, run the unit tests:
`./gradlew :domain-ee:test`


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
